cas-server-support-jpa-ticket-registry/src/main/java/org/jasig/cas/ticket/registry/JpaTicketRegistry.java

```diff
@@ -168,8 +171,10 @@ public Ticket getTicket(final String ticketId) {
      */
     private Ticket getRawTicket(final String ticketId) {
         try {
-            if (ticketId.startsWith(this.ticketGrantingTicketPrefix)) {
-                return entityManager.find(TicketGrantingTicketImpl.class, ticketId);
+            for(final String prefix : this.ticketGrantingTicketPrefixes) {
+                if (ticketId.startsWith(prefix)) {
+                    return entityManager.find(TicketGrantingTicketImpl.class, ticketId);
```

Comment:
I can add a comment to clarify, this would account for any `@Entity` that inherits from `TicketGrantingTicketImpl` because they are stored in the same DB table and JPA will seamlessly find any object subclasses.

This logic would have to change if we ever separate PGTs out to a different table.